realsense_ros/src/rs_d435.cpp

```diff
@@ -77,19 +77,22 @@ void RealSenseD435::publishTopicsCallback(const rs2::frame & frame)
     auto frame = frameset.get_infrared_frame(2);
     publishImageTopic(frame, t);
   }
-  if (align_depth_ && (aligned_depth_image_pub_->get_subscription_count() > 0 || aligned_depth_info_pub_->get_subscription_count() > 0)) {
+   if ((enable_pointcloud_ && pointcloud_pub_->get_subscription_count() > 0) || 
```

Comment:
I don't quite understand this line. Why do you mix the point cloud and aligned depth into one condition clause? I suggest you separate them, because the point cloud that is not aligned to color will be also supported in future.